Tasks/Common/packaging-common/universal/ArtifactToolUtilities.ts

```diff
@@ -156,34 +156,23 @@ export async function getHighestPackageVersionFromFeed(serviceUri: string, acces
     }
 
     // Getting url for feeds version API
-    const packageUrl = await new Promise<string>((resolve, reject) => {
-        var getVersioningDataPromise = feedConnection.vsoClient.getVersioningData(ApiVersion, PackagingAreaName, PackageAreaId, routeValues, {packageNameQuery: packageName, protocolType: "upack", includeDeleted: "true", includeUrls: "false"});
-        getVersioningDataPromise.then((result) => {
-            return resolve(result.requestUrl);
-        });
-        getVersioningDataPromise.catch((error) => {
-            return reject(error);
-        });
-    });
-
-    const versionResponse = await new Promise<string>((resolve, reject) => {
-        let responsePromise = feedConnection.rest.get(packageUrl);
-        responsePromise.then((result) => {
-            if (result.result['count'] === 0){
-                return resolve("0.0.0");
-            }
-            else{
-                result.result['value'].forEach((element) => {
-                    if (element.name === packageName.toLowerCase()){
-                        return resolve(element.versions[0].version);
-                    }
-                });
-            }
-        });
-        responsePromise.catch((error) => {
-            return reject(error);
-        });
-    });
-
-    return versionResponse;
+    const data = await feedConnection.vsoClient.getVersioningData(ApiVersion, PackagingAreaName, PackageAreaId, routeValues, {packageNameQuery: packageName, protocolType: "upack", includeDeleted: "true", includeUrls: "false"});
+    
+    console.log('Resolved Route: ' + data.requestUrl);
+
+    const result = await feedConnection.rest.get(data.requestUrl);
+    if(result.result != null) {
+        if (result.result['count'] == null || result.result['count']  === 0){
```

Comment:
```suggestion
        if (result.result['count'] === null || result.result['count']  === 0){
```

Reply:
alternatively, you could just see if result.result['count'] is "falsy" 
```suggestion
        if (!result.result['count']){
```